hearts/hearts.py

```diff
@@ -87,65 +94,64 @@ def __eq__(self, other):
         return {card.serialize() for card in self} == {card.serialize() for card in other}
 
 
-class Error(Exception):
-    pass
-
-
-class CardError(Error):
-    pass
-
-
-class HeartsError(Error):
-    pass
-
-
-class EarlyDumpError(Error):
-    ''' For when a player tries to dump on the first round'''
-    pass
-
-
 class Round(object):
     def __init__(self, players):
         '''
             A Round object tracks the state of a given round.
         '''
         self.players = players   # List of players in seated order
         self.hands = dict()      # Player -> Hand
-        self.tricks = []         # Should all 13 tricks be initialized at the beginning?
-        self.turn = None         # FIXME: implement turn counter
+        self.tricks = []
+        self.turn_counter = 0
         self.hearts_broken = False
-        self.who_starts = None   # FIXME: find the two of clubs
 
         self.deal()
+        self.set_turn_counter()
 
     def deal(self):
         deck = CARDS
         shuffle(deck)
         for n in range(4):
-            self.hands[self.players[n]] = Hand(deck[13*n:13*(n+1)])
+            start_hand = Hand(deck[13*n:13*(n+1)])
+            start_hand.hand_sort()
+            self.hands[self.players[n]] = start_hand
+
+    def set_turn_counter(self):
+        for index in range(4):
+            if Card('2', 'c') in self.hands[self.players[index]]:
+                self.turn_counter = index
 
     def can_follow_suit(self, player, trick):
         hand = self.hands[player]
         return hand.has_suit(trick.suit)
 
     def is_valid_lead(self, player, card):
-        if card.suit == 'h' and not self.hearts_broken:
-            if not all(card.suit == 'h' for card in self.hands[player]):
-                raise HeartsError
-            else:  # Player has no other available suit to lead with
-                self.hearts_broken = True
-        return
+        '''
+           Return a boolean for whether the player can use the given card
+           to start the trick.
+        '''
+        if len(self.tricks) == 0:
+            return card == Card('2', 'c')
+        elif card.suit == 'h' and not self.hearts_broken:
+            return self.hands[player].is_only_hearts()
+        else:
+            return True
 
     def is_valid_follow(self, player, trick, card):
-        if card.suit == trick.suit:
-            return
+        '''
+            Return a boolean for whether the player
+            can use the given card to follow the trick
+        '''
+        if self.can_follow_suit(player, trick):
+            return card.suit == trick.suit
+        elif len(self.tricks) == 1:
+            player_hand = self.hands[player]
+            return player_hand.is_only_hearts() or player_hand.has_only_hearts_and_Qs()
```

Comment:
This is not the correct condition. In particular, it doesn't depend on the input card. Maybe add a method `card.is_worth_points()`?